src/linalg/matmul.jl

```diff
@@ -47,7 +47,9 @@ end
 # scale!
 # scale
 # vecdot
-# dot
+@primitive dot(x1, x2),dy,y  dy*x2  dy*x1
+gradcheckN(dot, rand(3,2), rand(3,2))
```

Comment:
Could you please add test ``` addtestN(:dot, rand(3,2), rand(2,3))``` for your implementation

Reply:
good catch